File: python/cart_shipment_groups.py
import sdk.python.entities
from sdk.python.entities import Property


class CartShipmentGroup(sdk.python.entities.Entity):

    resource = "/cart_shipment_groups/"
    json_name = "cartShipmentGroup"

    id = Property(int)
    items = Property("sdk.python.cart_items.CartItem")
    quotes = Property("sdk.python.cart_shipment_quotes.CartShipmentQuote")
    selectedQuote = Property(
        "sdk.python.cart_shipment_quotes.CartShipmentQuote"
    )


class CartShipmentGroups(sdk.python.entities.Resource):

    entity_class = CartShipmentGroup
    json_name = "cartShipmentGroups"


cart_shipment_groups = CartShipmentGroups()

File: php/src/cart_shipment_groups.php
<?php

declare(strict_types=1);

require_once 'entity.php';

class CartShipmentGroup extends Entity
{
    public static $json_name = 'cart_shipment_group';
    public static $resource = '/cart_shipment_groups/';


    public function __construct()
    {
        $this->json_property('id', 'integer');

        $this->json_property('cartItems', 'CartItem', $many = True,
            $recursive = True);
        $this->json_property('quotes', 'CartShipmentQuote', $many = True,
            $recursive = True);
        $this->json_property('selectedQuote', 'CartShipmentQuote', $many = False,
            $recursive = True);
    }
}
